Classes/Units/Controllers/Inventory.js

```diff
@@ -65,13 +65,42 @@ class Inventory {
 		if(slot === false)
 			return false;
 
+		var effectiveGoldCost = Item.GoldCost
+
+		if(Item.from)
```

Comment:
Should be together with if at line 60.
At the moment if we will have full inventory with items that can be builded to item we want to buy, slot will be false so line:66 will return false instead of builiding item from them.

Also `this.getEmptySlot()` is used twice unnecessarily.